### PR TITLE
Add more granular schema for stats in options schema

### LINKED ISSUE #4264: context: false in stats throws
Following webpack config throws an error:

```js
var webpack = require('webpack');

/** @type {webpack.Configuration} */
module.exports = {
    entry: {
        main: './index.js'
    },
    output: {
        path: 'dist',
        filename: '[name].js'
    },
    stats: {
        context: false, // Removing this fixes it
        colors: true,
        chunkOrigins: false,
        hash: false,
        version: false,
        timings: false,
        assets: false,
        chunks: false,
        chunkModules: false,
        modules: false,
        children: false,
        cached: false,
        reasons: false,
        source: false,
        errorDetails: false,
    },
}
```

**Error**

```

> test@1.0.0 test /Users/mazimi/Desktop/test
> webpack

/Users/mazimi/Desktop/test/node_modules/webpack/lib/RequestShortener.js:11
		directory = directory.replace(/\\/g, "/");
		                      ^

TypeError: directory.replace is not a function
    at new RequestShortener (/Users/mazimi/Desktop/test/node_modules/webpack/lib/RequestShortener.js:11:25)
    at Stats.toJson (/Users/mazimi/Desktop/test/node_modules/webpack/lib/Stats.js:35:28)
    at Stats.toString (/Users/mazimi/Desktop/test/node_modules/webpack/lib/Stats.js:355:20)
    at compilerCallback (/Users/mazimi/Desktop/test/node_modules/webpack/bin/webpack.js:340:31)
    at /Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:256:15
    at Compiler.emitRecords (/Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:351:37)
    at /Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:249:12
    at /Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:344:11
    at next (/Users/mazimi/Desktop/test/node_modules/tapable/lib/Tapable.js:154:11)
    at Compiler.compiler.plugin (/Users/mazimi/Desktop/test/node_modules/webpack/lib/performance/SizeLimitsPlugin.js:99:4)
    at Compiler.applyPluginsAsyncSeries1 (/Users/mazimi/Desktop/test/node_modules/tapable/lib/Tapable.js:158:13)
    at Compiler.afterEmit (/Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:341:8)
    at Compiler.<anonymous> (/Users/mazimi/Desktop/test/node_modules/webpack/lib/Compiler.js:336:14)
    at /Users/mazimi/Desktop/test/node_modules/async/dist/async.js:356:16
    at iteratorCallback (/Users/mazimi/Desktop/test/node_modules/async/dist/async.js:936:13)
    at /Users/mazimi/Desktop/test/node_modules/async/dist/async.js:840:16
    at /Users/mazimi/Desktop/test/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqWrap.oncomplete (fs.js:123:15)
npm ERR! Test failed.  See above for more details.
```

### PR DESCRIPTION
**Pull request by mohsen1**
Fixes #4264

### CODE REVIEW

File: schemas/webpackOptionsSchema.json [hunk 1]
```diff
@@ -873,7 +893,77 @@
       "description": "Used by the webpack CLI program to pass stats options.",
       "anyOf": [
         {
-          "type": "object"
+          "type": "object",
+          "properties": {
+            "context": {
+              "type": "string",
+              "description": "context directory for request shortening"
```
Comment (sokra) [inline schemas/webpackOptionsSchema.json]: add `absolutePath: true` here

Comment (mohsen1) [inline schemas/webpackOptionsSchema.json]: Done. Is that custom to this JSON schema? Never seen that before.

Comment (sokra) [inline schemas/webpackOptionsSchema.json]: Yep. It's custom.

**Comment by sokra** [comment]
Thanks

**Pull request merged**